From caf1c127aaac926bcbdb06c8db519848716c5070 Mon Sep 17 00:00:00 2001
From: Stefan Agner <stefan.agner@toradex.com>
Date: Tue, 10 Dec 2019 13:57:11 +0100
Subject: [PATCH] fix cross-compilation

Make sure that libdrm found by pkg-config from the sysroot is used.
This likely will be fixed by future meson builds:
https://github.com/scivision/meson/commit/7a65ee41a8390fd50ac723feb8c5452d2b23d8c8

Upstream-Status: Inappropriate [embedded specific]
Signed-off-by: Stefan Agner <stefan.agner@toradex.com>
---
 meson.build       | 2 +-
 meson_options.txt | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)
 create mode 100644 meson_options.txt

diff --git a/meson.build b/meson.build
index 6241cf0..2a3472d 100644
--- a/meson.build
+++ b/meson.build
@@ -25,7 +25,7 @@ config_h = configure_file(
 )
 
 python3 = import('python').find_installation()
-drm_fourcc_h_path = join_paths(libdrm.get_pkgconfig_variable('includedir'), 'libdrm/drm_fourcc.h')
+drm_fourcc_h_path = get_option('pkgconfig-sysroot-path') + join_paths(libdrm.get_pkgconfig_variable('includedir'), 'libdrm/drm_fourcc.h')
 fourcc_py = join_paths(meson.source_root(), 'fourcc.py')
 
 tables_h = custom_target('tables_h',
diff --git a/meson_options.txt b/meson_options.txt
new file mode 100644
index 0000000..ba56d6d
--- /dev/null
+++ b/meson_options.txt
@@ -0,0 +1,3 @@
+option('pkgconfig-sysroot-path', type: 'string',
+    description: 'Specify a sysroot path to prepend to pkgconfig output (useful when cross-compiling)'
+)
-- 
2.24.0

